crates/relayer-components/src/chain/traits/queries/unreceived_packets.rs

```diff
@@ -0,0 +1,67 @@
+use async_trait::async_trait;
+
+use crate::chain::traits::types::ibc::HasIbcChainTypes;
+use crate::chain::traits::types::packet::HasIbcPacketTypes;
+use crate::core::traits::error::HasErrorType;
+use crate::std_prelude::*;
+
+#[async_trait]
+pub trait CanQueryUnreceivedPacketSequences<Counterparty>:
+    HasIbcChainTypes<Counterparty> + HasIbcPacketTypes<Counterparty> + HasErrorType
+where
+    Counterparty: HasIbcChainTypes<Self>,
+{
+    /// Given a list of counterparty commitment sequences,
+    /// return a filtered list of sequences which the chain
+    /// has not received the packet from the counterparty chain.
+    async fn query_unreceived_packet_sequences(
+        &self,
+        channel_id: &Self::ChannelId,
+        port_id: &Self::PortId,
+        sequences: &[Counterparty::Sequence],
+    ) -> Result<Vec<Counterparty::Sequence>, Self::Error>;
+}
+
+#[async_trait]
+pub trait SendPacketsFromSequencesQuerier<Chain, Counterparty>
+where
+    Chain: HasIbcChainTypes<Counterparty> + HasIbcPacketTypes<Counterparty> + HasErrorType,
+    Counterparty: HasIbcChainTypes<Chain>,
+{
+    /// Given a list of sequences, a channel and port will query a list of outgoing
+    /// packets which have not been relayed.
+    async fn query_unreceived_packets(
+        &self,
+        channel_id: &Chain::ChannelId,
+        port_id: &Chain::PortId,
+        counterparty_channel_id: &Counterparty::ChannelId,
+        counterparty_port_id: &Counterparty::PortId,
+        sequences: &[Chain::Sequence],
+        // The height is given to query the packets from a specific height.
+        // This height should be the same as the query height from the
+        // `CanQueryPacketCommitments` made on the same chain.
+        height: &Chain::Height,
+    ) -> Result<Vec<Chain::OutgoingPacket>, Chain::Error>;
+}
+
+#[async_trait]
+pub trait CanQuerySendPacketsFromSequences<Counterparty>:
+    HasIbcChainTypes<Counterparty> + HasIbcPacketTypes<Counterparty> + HasErrorType
+where
+    Counterparty: HasIbcChainTypes<Self>,
+{
+    /// Given a list of sequences, a channel and port will query a list of outgoing
+    /// packets which have not been relayed.
+    async fn query_unreceived_packets(
```

Comment:
The name of the method should also change to `query_send_packets_from_sequences`. Also, let's move this trait to a separate file like `send_packets.rs`.